### ISSUE TITLE
LayerNorm does not normalize to unitary variance.

### ISSUE THREAD

**Issue by alledoubleL** [pytorch/pytorch#84356]
### 🐛 Describe the bug

Hi!

LayerNorm should normalize to zero mean and unitari variance the input and then linearly rescale everything with $\gamma$ and $\beta$ parameters (look the [official doc](https://pytorch.org/docs/stable/generated/torch.nn.LayerNorm.html#torch.nn.LayerNorm)).

Anyway if we do a simple test we obtain that the output has not unitary variance:

```python
import torch
from torch.nn.parameter import Parameter

s1 = torch.tensor([[0., 2.], [0., 4.], [-1., 5.]])

# Normalize each row with torch functions:
zero_mean = s1 - torch.mean(s1, axis=1).unsqueeze(1)
print(zero_mean)
>> tensor([[-1.,  1.],
           [-2.,  2.],
           [-3.,  3.]])

unit_var = zero_mean / (torch.std(zero_mean, axis=1).unsqueeze(1))
print(unit_var)
>> tensor([[-0.7071,  0.7071],
           [-0.7071,  0.7071],
           [-0.7071,  0.7071]])

print(torch.std(unit_var, axis=1))
>> tensor([1.0000, 1.0000, 1.0000])

# Normalize with LayerNorm
normalization = torch.nn.LayerNorm(2)

normalization(s1)
>> tensor([[-1.0000,  1.0000],
           [-1.0000,  1.0000],
           [-1.0000,  1.0000]], grad_fn=<NativeLayerNormBackward0>)
```

The output is clearly not normalized! Note that this is not due to the $\gamma$ and $\beta$ parameters as they are correctly initialized:

```python
print(normalization.weight) # Gamma parameter
>> Parameter containing:
   tensor([1., 1.], requires_grad=True)

print(normalization.bias) # Beta parameter
>> Parameter containing:
   tensor([0., 0.], requires_grad=True)
```

### Versions

yTorch version: 1.11.0+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.4 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.31

Python version: 3.8.10 (default, Jun 22 2022, 20:18:18)  [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.13.0-28-generic-x86_64-with-glibc2.29
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1650 Ti
Nvidia driver version: 470.141.03
cuDNN version: Probably one of the following:
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn.so.8
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_adv_train.so.8
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_ops_train.so.8
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.21.6
[pip3] torch==1.11.0
[pip3] torchmetrics==0.9.2
[pip3] torchvision==0.12.0
[conda] Could not collect

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are

**Comment by alledoubleL** [comment]
It seems that instead of using L2 normalization it uses L1. Maybe?

**Comment by ngimel** [comment]
As the docs say https://pytorch.org/docs/stable/generated/torch.nn.LayerNorm.html?highlight=layernorm#torch.nn.LayerNorm, biased estimated for variance is used, so results are as expected
```
In [8]: s1 = torch.tensor([[0., 2.], [0., 4.], [-1., 5.]])

In [9]: normalization = torch.nn.LayerNorm(2)
   ...: 
   ...: out=normalization(s1)

In [10]: out.var(dim=-1, unbiased=False)
Out[10]: tensor([1.0000, 1.0000, 1.0000], grad_fn=<VarBackward0>)
```
